cylc/flow/etc/job.sh

```diff
@@ -140,6 +150,46 @@ cylc__job__main() {
     exit 0
 }
 
+###############################################################################
+# Run user scripts in subshell to protect cylc job script from interference.
+# Getting signal traps to run immediately requires waiting on a background
+# process.
+cylc__job__run_user_scripts() {
+    typeset func_name=
+    for func_name in 'env_script' 'user_env' 'pre_script' 'script' 'post_script'; do
+        cylc__job__run_inst_func "${func_name}"
+    done
+}
+
+###############################################################################
+# Disable selected or all (if no arguments given) fail traps.
+# Globals:
+#   CYLC_FAIL_SIGNALS
+cylc__job__disable_fail_signals() {
```

Comment:
I figured it's cleaner to add a function for this than directly muck around with CYLC_FAIL_SIGNALS variable in the tests.